spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/factory/rewrite/ModifyRequestBodyGatewayFilterFactory.java

```diff
@@ -110,22 +111,17 @@ public String toString() {
 		};
 	}
 
-	ServerHttpRequestDecorator decorate(ServerWebExchange exchange, HttpHeaders headers,
+	ServerHttpRequestDecorator decorate(ServerHttpRequest request, HttpHeaders headers,
 			CachedBodyOutputMessage outputMessage) {
-		return new ServerHttpRequestDecorator(exchange.getRequest()) {
+		return new ServerHttpRequestDecorator(request) {
 			@Override
 			public HttpHeaders getHeaders() {
-				long contentLength = headers.getContentLength();
 				HttpHeaders httpHeaders = new HttpHeaders();
-				httpHeaders.putAll(super.getHeaders());
```

Comment:
All changes in this class need to be reverted except here. This should be:
```
				httpHeaders.putAll(headers);
```
instead of the call to `super.getHeaders()`.

Reply:
I reflected your comment.
Thank you.